src/components/mainContainer/mainContainer.ts

```diff
@@ -0,0 +1,29 @@
+import RootComponent from '@common/rootComponent';
+import { IRootComponent } from '@common/types';
+import ModalWindow from '../modalWindow/modalWindow';
+import { findComponent } from '@common/utils';
+import { BehaviorSubject, fromEvent } from 'rxjs';
+
+class MainContainer extends RootComponent<HTMLDivElement> {
+    readonly count: HTMLDivElement;
+    readonly openButton: HTMLButtonElement;
+    readonly modalWindow: ModalWindow;
+
+    constructor(props: IRootComponent) {
+        super(props);
+
+        this.modalWindow = new ModalWindow(findComponent('modal-window'));
+        this.openButton = this.node.querySelector(`.${this.name}__openButton`);
+        this.count = this.node.querySelector(`.${this.name}__count`);
+
+        fromEvent(this.openButton, 'click').subscribe((target) => {
+            this.modalWindow.toggleW();
+        });
+
+        this.modalWindow.currentValue$.subscribe((target) => {
+            this.count.textContent = `${target}`;
+        });
```

Comment:
Будет хорошей привычкой делать обработчики отдельными методами вместо использования  анонимных функций  
Такой подход позволит не хранить тело функции в конструкторе класса и в большинстве случаев улучшит чтение кода
```suggestion
        fromEvent(this.openButton, 'click').subscribe(this.clickHandler);
        this.modalWindow.currentValue$.subscribe(this.updateContent);
    }
    
    private clickHandler = () => {
        this.modalWindow.toggleW();
    }
    
    private updateContent = (target: number) => {
        this.count.textContent = `${target}`;
    }
```